src/main/java/nl/hsac/fitnesse/fixture/slim/web/BrowserTest.java

```diff
@@ -2001,13 +2001,43 @@ protected String savePageSource(String fileName, String linkText) {
     }
 
     /**
-     * Takes screenshot from current page
+     * Takes screenshot of page as currently visible
      * @param basename filename (below screenshot base directory).
      * @return location of screenshot.
      */
     public String takeScreenshot(String basename) {
+        return takeScreenshotWith(
+            basename,
+            name ->
+                createScreenshot(name)
+        );
+    }
+
+    /**
+     * Takes screenshot of whole page
+     * @param basename filename (below screenshot base directory).
+     * @return location of screenshot.
+     */
+    public String fullPageScreenshot(String basename) {
+        return takeScreenshotWith(
+            basename,
+            name -> 
+                getSeleniumHelper()
+                    .fullPageScreenshot(
+                        getScreenshotBasename(name)
+                    )
+        );
+    }
+
+    /**
+     * Takes screenshot using the passed maker function
+     * @param basename filename (below screenshot base directory).
+     * @param maker    function for taking the screenshot, should return the screenshot location.
+     * @return location of screenshot.
+     */
+    private String takeScreenshotWith(String baseName, Function<String, String> maker) {
```

Comment:
Please make protected, so people can override in subclasses, should they want to